src/components/portfolio/Portfolio.js

```diff
@@ -22,7 +25,108 @@ class Portfolio extends Component {
     )
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      stockDetails: [],
+      cryptoDetails: [],
+      isLoading: true,
+      showStart: false
+    };
+    this.onDelete = this.onDelete.bind(this);
+  }
+
+  onDelete(index) {
+    let stateObj = this.state;
+    let stockDetails = stateObj.stockDetails;
+    stockDetails.splice(index, 1);
+    stateObj.stockDetails = stockDetails;
+    this.setState(stateObj);
+    let portfolioDetails = {};
+    portfolioDetails.stockDetails = stockDetails;
+    AsyncStorage.setItem("portfolioDetails", JSON.stringify(portfolioDetails));
+  }
+
+  componentDidMount() {
+    /*     let portfolioDetails = {};
+    let { stockDetails = [] } = [];
+    const stockObj = {
+      name: 'sds',
+      symbol: 'sdsd',
+      exchDisp: 'sdsdsd',
+      quantity: 'sdsd',
+      price: 'sdsd'
+    };
+
+    stockDetails.push(stockObj);
+    portfolioDetails.stockDetails = stockDetails;
+    AsyncStorage.setItem("portfolioDetails",JSON.stringify(portfolioDetails));
+ */
+    AsyncStorage.getItem("portfolioDetails").then(response => {
+      let portfolioDetails = JSON.parse(response);
```

Comment:
A concise way
```
       const portfolioDetails = JSON.parse(response) || {};
       const { 
           stockDetails = [],
           cryptoDetails = []
         } = portfolioDetails;
 
       this.setState({
          stockDetails,
          cryptoDetails,
          isLoading: false,
          showStart: !!(stockDetails.length || cryptoDetails.length)
       });

```
See you really do not need to `let` in many cases. If you try use `const` it will make you rethink every time you want to reassign the value to the variable. ESXXXX object and other features lets you write things in a concise way.